File: CPP [2001-2500]/2109.adding_spaces_to_a_string [1].cpp
class Solution {
public:
    // T.C.=O(n + m), S.C.=O(1)
    // n: length of 's', m: size of spaces[]
    string addSpaces(string s, vector<int>& spaces) {
        string ans = "";

        for(int i=0, j=0; i<s.length(); i++)
        {
            if(j<spaces.size() && spaces[j] == i)
            {
                ans += " ";
                j++;
            }
            ans += s[i];
        }

        return ans;
    }
};

File: Javascript/2109.adding-spaces-to-a-string [1].js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function(s, spaces) {
    let ans = s.substring(0, spaces[0])

    for(let i=1; i<spaces.length; i++){
        ans += ' ' + s.substring(spaces[i-1], spaces[i])
    }

    ans += ' ' + s.substring(spaces[spaces.length - 1])

    return ans
};